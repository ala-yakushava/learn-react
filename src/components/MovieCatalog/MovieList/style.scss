@import "../../../styles/theme";

.MovieList {
  &_lead {
    font-size: 24px;
  }

  &_list {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: repeat(3, min-content);
    grid-gap: 60px;
  }
}
