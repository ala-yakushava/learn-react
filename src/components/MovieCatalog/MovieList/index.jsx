import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';
import MovieCard from '../MovieCard';

const MovieList = ({ movies, genres }) => (
  <section className="MovieList">
    <p className="MovieList_lead">
      { movies.length }
      {' '}
      movies found
    </p>
    <div className="MovieList_list">
      { movies.map((movie) => <MovieCard movie={movie} genres={genres} key={movie.id} />) }
    </div>
  </section>
);

MovieList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object),
  genres: PropTypes.arrayOf(PropTypes.object),
};

MovieList.defaultProps = {
  movies: [],
  genres: [],
};

export default MovieList;
