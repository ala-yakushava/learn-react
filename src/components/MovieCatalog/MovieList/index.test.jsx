import React from 'react';
import renderer from 'react-test-renderer';

import MovieList from './index';

describe('Components - MovieList', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<MovieList />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
