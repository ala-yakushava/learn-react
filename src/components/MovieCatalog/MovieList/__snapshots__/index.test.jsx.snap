// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - MovieList should match snapshot 1`] = `
<section
  className="MovieList"
>
  <p
    className="MovieList_lead"
  >
    0
     movies found
  </p>
  <div
    className="MovieList_list"
    data-testid="movie-list"
  />
</section>
`;
